src/main/solution_space/serialized_code.py

```diff
@@ -106,14 +106,19 @@ def get_anon_files(self, filter_anon_trees: Callable[[ast.AST], bool] = (lambda
     def get_canon_file(self) -> str:
         return self.__get_file_by_tree(self._canon_tree)
 
-    def __create_files_for_trees(self) -> None:
-        self.__create_file_for_tree(self._canon_tree, TREE_TYPE.CANON.value)
+    def rewrite_code_info(self, new_folder_with_files: str) -> None:
+        self._folder_with_files = new_folder_with_files
+        self.__create_files_for_trees(to_rewrite=True)
+
+    def __create_files_for_trees(self, to_rewrite: bool = False) -> None:
+        self.__create_file_for_tree(self._canon_tree, TREE_TYPE.CANON.value, to_rewrite=to_rewrite)
         for i, anon_tree in enumerate(self._anon_trees):
-            self.__create_file_for_tree(anon_tree, f'{TREE_TYPE.ANON.value}_{i}')
+            self.__create_file_for_tree(anon_tree, f'{TREE_TYPE.ANON.value}_{i}', to_rewrite=to_rewrite)
 
     # If file exists already in graph folder, we don't want to override it
-    def __create_file_for_tree(self, tree: ast.AST, str_tree_type: str) -> str:
-        if self._file_by_tree_dict.get(tree) is not None:
+    def __create_file_for_tree(self, tree: ast.AST, str_tree_type: str,
+                               to_rewrite: bool = False) -> str:
```

Comment:
There is a little difference between ['rewrite'](https://dictionary.cambridge.org/dictionary/english/rewrite) and ['overwrite'](https://dictionary.cambridge.org/dictionary/english/overwrite). It's better to use 'overwrite' speaking about 'changing content of files'. So let's rename it to 'to_overwrite'.